Strapi 5 (beta 17) creating a simple component breaks the server

## Bug report

### Required System information

- Node.js version: 20.14
- NPM version: 10.7
- Strapi version: 5.0.0-beta.17
- Database: SQLite
- Operating system: Windows
- Is your project Javascript or Typescript: Typescript

<!-- Did you know you can run `yarn strapi report` or `npm run strapi report` to get this information easier? -->

### Describe the bug

Creating a simple component with 2 text fields causes an error:
\ Loading Strapi[ERROR]  There seems to be an unexpected error, try again with --debug for more information

┌─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│                                                                                                                                         │
│   TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received undefined                                      │
│       at Object.join (node:path:433:7)                                                                                                  │
│       at D:\Code\Droplet\Strapi\betalatest\node_modules\@strapi\core\dist\loaders\components.js:17:31                                   │
│       at Array.forEach (<anonymous>)                                                                                                    │
│       at D:\Code\Droplet\Strapi\betalatest\node_modules\@strapi\core\dist\loaders\components.js:14:32                                   │
│       at Array.reduce (<anonymous>)                                                                                                     │
│       at loadComponents (D:\Code\Droplet\Strapi\betalatest\node_modules\@strapi\core\dist\loaders\components.js:13:39)                  │
│       at async Promise.all (index 5)                                                                                                    │
│       at async Module.loadApplicationContext (D:\Code\Droplet\Strapi\betalatest\node_modules\@strapi\core\dist\loaders\index.js:12:3)   │
│       at async Object.register (D:\Code\Droplet\Strapi\betalatest\node_modules\@strapi\core\dist\providers\registries.js:26:5)          │
│       at async Strapi.register (D:\Code\Droplet\Strapi\betalatest\node_modules\@strapi\core\dist\Strapi.js:322:7)                       │
│                                                                                                                         

### Steps to reproduce the behavior

1. npx create-strapi-app@beta betalatest --quickstart
2. Start the admin panel
3. Create a simple component with 2 text fields
4. Get the error on reload (see above)

### Expected behavior

This is a super basic operation and should just work

--- a/packages/core/core/src/utils/__tests__/filepath-to-prop-path.test.ts
+++ b/packages/core/core/src/utils/__tests__/filepath-to-prop-path.test.ts
@@ -37,11 +37,6 @@ describe('filePathToPropPath', () => {
       expect(filePathToPropPath('./config/test.js')).toEqual(['config', 'test']);
     });
 
-    test('Win32 Separators', () => {
-      expect(filePathToPropPath('config\\test.js')).toEqual(['config', 'test']);
-      expect(filePathToPropPath('.\\config\\test.js')).toEqual(['config', 'test']);
-    });
-
     test('Mixed Separators (win32 + posix)', () => {
       expect(filePathToPropPath('src\\config/test.js')).toEqual(['src', 'config', 'test']);
       expect(filePathToPropPath('.\\config/test.js')).toEqual(['config', 'test']);

--- a/packages/core/core/src/utils/__tests__/filepath-to-prop-path.test.ts
+++ b/packages/core/core/src/utils/__tests__/filepath-to-prop-path.test.ts
@@ -25,4 +25,26 @@ describe('filePathToPropPath', () => {
     expect(filePathToPropPath('./config/test.js', false)).toEqual(['config']);
     expect(filePathToPropPath('./config/test.key.js', false)).toEqual(['config', 'test']);
   });
+
+  describe('Separators', () => {
+    test('Win32 Separators', () => {
+      expect(filePathToPropPath('config\\test.js')).toEqual(['config', 'test']);
+      expect(filePathToPropPath('.\\config\\test.js')).toEqual(['config', 'test']);
+    });
+
+    test('Posix Separators', () => {
+      expect(filePathToPropPath('config/test.js')).toEqual(['config', 'test']);
+      expect(filePathToPropPath('./config/test.js')).toEqual(['config', 'test']);
+    });
+
+    test('Win32 Separators', () => {
+      expect(filePathToPropPath('config\\test.js')).toEqual(['config', 'test']);
+      expect(filePathToPropPath('.\\config\\test.js')).toEqual(['config', 'test']);
+    });
+
+    test('Mixed Separators (win32 + posix)', () => {
+      expect(filePathToPropPath('src\\config/test.js')).toEqual(['src', 'config', 'test']);
+      expect(filePathToPropPath('.\\config/test.js')).toEqual(['config', 'test']);
+    });
+  });
 });

--- a/packages/core/core/src/utils/filepath-to-prop-path.ts
+++ b/packages/core/core/src/utils/filepath-to-prop-path.ts
@@ -16,7 +16,7 @@ export const filePathToPropPath = (
     // Make sure it's lowercase
     fp.lowerCase,
     // Split the cleaned path by matching every possible separator (either "/" or "\" depending on the OS)
-    fp.split(new RegExp(`${path.win32.sep}|${path.posix.sep}`)),
+    fp.split(new RegExp(`${path.win32.sep}|${path.posix.sep}`, 'g')),
     // Make sure to remove leading '.' from the different path parts
     fp.map(fp.trimCharsStart('.')),
     // join + split in case some '.' characters are still present in different parts of the path

--- a/packages/core/core/src/utils/filepath-to-prop-path.ts
+++ b/packages/core/core/src/utils/filepath-to-prop-path.ts
@@ -31,7 +31,7 @@ export const filePathToPropPath = (
 };
 
 const removeRelativePrefix = (filePath: string) => {
-  return filePath.startsWith(`.\\${path.win32.sep}`) || filePath.startsWith(`.${path.posix.sep}`)
+  return filePath.startsWith(`.${path.win32.sep}`) || filePath.startsWith(`.${path.posix.sep}`)
     ? filePath.slice(2)
     : filePath;
 };

--- a/packages/core/core/src/utils/filepath-to-prop-path.ts
+++ b/packages/core/core/src/utils/filepath-to-prop-path.ts
@@ -9,7 +9,7 @@ export const filePathToPropPath = (
   useFileNameAsKey: boolean = true
 ): string[] => {
   const transform = fp.pipe(
-    // Remove the relative path prefix (posix or win32) and ".\" (win32) prefixes
+    // Remove the relative path prefixes: './' for posix (and some win32) and ".\" for win32
     removeRelativePrefix,
     // Remove the path metadata and extensions
     fp.replace(/(\.settings|\.json|\.js)/g, ''),

--- a/packages/core/core/src/utils/filepath-to-prop-path.ts
+++ b/packages/core/core/src/utils/filepath-to-prop-path.ts
@@ -1,3 +1,4 @@
+import path from 'node:path';
 import _ from 'lodash';
 
 /**
@@ -9,7 +10,7 @@ export const filePathToPropPath = (filePath: string, useFileNameAsKey = true) =>
   const prop = cleanPath
     .replace(/(\.settings|\.json|\.js)/g, '')
     .toLowerCase()
-    .split('/')
+    .split(path.sep)
     .map((p) => _.trimStart(p, '.'))
     .join('.')
     .split('.');

--- a/packages/core/core/src/utils/filepath-to-prop-path.ts
+++ b/packages/core/core/src/utils/filepath-to-prop-path.ts
@@ -13,10 +13,11 @@ export const filePathToPropPath = (
     removeRelativePrefix,
     // Remove the path metadata and extensions
     fp.replace(/(\.settings|\.json|\.js)/g, ''),
-    // Make sure it's lowercase
-    fp.lowerCase,
+    // Transform to lowercase
+    // Note: We're using fp.toLower instead of fp.lowercase as the latest removes special characters such as "/"
+    fp.toLower,
     // Split the cleaned path by matching every possible separator (either "/" or "\" depending on the OS)
-    fp.split(new RegExp(`${path.win32.sep}|${path.posix.sep}`, 'g')),
+    fp.split(new RegExp(`[${path.win32.sep}|${path.posix.sep}]`, 'g')),
     // Make sure to remove leading '.' from the different path parts
     fp.map(fp.trimCharsStart('.')),
     // join + split in case some '.' characters are still present in different parts of the path

--- a/packages/core/core/src/utils/filepath-to-prop-path.ts
+++ b/packages/core/core/src/utils/filepath-to-prop-path.ts
@@ -17,7 +17,7 @@ export const filePathToPropPath = (
     // Note: We're using fp.toLower instead of fp.lowercase as the latest removes special characters such as "/"
     fp.toLower,
     // Split the cleaned path by matching every possible separator (either "/" or "\" depending on the OS)
-    fp.split(new RegExp(`[${path.win32.sep}|${path.posix.sep}]`, 'g')),
+    fp.split(new RegExp(`[\\${path.win32.sep}|${path.posix.sep}]`, 'g')),
     // Make sure to remove leading '.' from the different path parts
     fp.map(fp.trimCharsStart('.')),
     // join + split in case some '.' characters are still present in different parts of the path
@@ -31,7 +31,7 @@ export const filePathToPropPath = (
 };
 
 const removeRelativePrefix = (filePath: string) => {
-  return filePath.startsWith(`.${path.win32.sep}`) || filePath.startsWith(`.${path.posix.sep}`)
+  return filePath.startsWith(`.\\${path.win32.sep}`) || filePath.startsWith(`.${path.posix.sep}`)
     ? filePath.slice(2)
     : filePath;
 };

--- a/packages/core/core/src/utils/filepath-to-prop-path.ts
+++ b/packages/core/core/src/utils/filepath-to-prop-path.ts
@@ -1,19 +1,36 @@
 import path from 'node:path';
-import _ from 'lodash';
+import fp from 'lodash/fp';
 
 /**
  * Returns a path (as an array) from a file path
  */
-export const filePathToPropPath = (filePath: string, useFileNameAsKey = true) => {
-  const cleanPath = filePath.startsWith('./') ? filePath.slice(2) : filePath;
+export const filePathToPropPath = (
+  entryPath: string,
+  useFileNameAsKey: boolean = true
+): string[] => {
+  const transform = fp.pipe(
+    // Remove the relative path prefix (posix or win32) and ".\" (win32) prefixes
+    removeRelativePrefix,
+    // Remove the path metadata and extensions
+    fp.replace(/(\.settings|\.json|\.js)/g, ''),
+    // Make sure it's lowercase
+    fp.lowerCase,
+    // Split the cleaned path by matching every possible separator (either "/" or "\" depending on the OS)
+    fp.split(new RegExp(`${path.win32.sep}|${path.posix.sep}`)),
+    // Make sure to remove leading '.' from the different path parts
+    fp.map(fp.trimCharsStart('.')),
+    // join + split in case some '.' characters are still present in different parts of the path
+    fp.join('.'),
+    fp.split('.'),
+    // Remove the last portion of the path array if the file name shouldn't be used as a key
+    useFileNameAsKey ? fp.identity : fp.slice(0, -1)
+  );
 
-  const prop = cleanPath
-    .replace(/(\.settings|\.json|\.js)/g, '')
-    .toLowerCase()
-    .split(path.sep)
-    .map((p) => _.trimStart(p, '.'))
-    .join('.')
-    .split('.');
+  return transform(entryPath) as string[];
+};
 
-  return useFileNameAsKey === true ? prop : prop.slice(0, -1);
+const removeRelativePrefix = (filePath: string) => {
+  return filePath.startsWith(`.${path.win32.sep}`) || filePath.startsWith(`.${path.posix.sep}`)
+    ? filePath.slice(2)
+    : filePath;
 };

